pallene/checker.lua

```diff
@@ -465,6 +465,37 @@ function FunChecker:check_stat(stat)
             stat.decl._name = self.p.symbol_table:find_symbol(stat.decl.name)
             self:check_stat(stat.block)
         end)
+        
+    elseif tag == "ast.Stat.ForIn" then
+        stat.exp = self:check_exp_synthesize(stat.exp)
+
+        if stat.exp._tag ~= "ast.Exp.CallFunc" then
+            type_error(stat.exp.loc,
+                "expected ipairs function but found %s in for-loop",
+                types.tostring(stat.exp._type),
+                stat.exp.name)
+        end
+
+        if stat.exp.exp._tag == "ast.Exp.Var" and
+           stat.exp.exp.var._name.name ~= "ipairs" and
+           stat.exp.exp.var._name._tag == "checker.Name.Builtin"
+        then
+            type_error(stat.exp.exp.loc, 
+                "expected ipairs function but found %s in for-loop",
+                stat.exp.exp.var.name)
```

Comment:
The error message in lines 473 and 483 are the same. This kind of duplication sounds suspicious. Do you think we should be using separate error messages, or that the code should be refactored so the error message is not repeated?